src/edtr-io/plugins/paste-hack/editor.tsx

```diff
@@ -0,0 +1,84 @@
+
+export const PasteHackEditor: React.FunctionComponent<PasteHackPluginProps> = (
+  props
+) => {
+  const store = useScopedStore()
+
+  function replaceWithStateString() {
+    console.log('starting replace')
+
+    // TODO: Use io-ts here
+    const content = JSON.parse(dummyContent) as {
+      plugin: 'rows'
+      state: Document[]
+    }
+
+    const parentPlugin = getParent(props.id)(store.getState())
+
+    if (parentPlugin == null) return
+
+    const documents = content.plugin === 'rows' ? content.state : [content]
+
+    console.log(store.getState())
+    console.log(serializeDocument(parentPlugin.id)(store.getState()))
+
+    for (const document of documents) {
+      store.dispatch(
+        insertChildBefore({
+          parent: parentPlugin.id,
+          sibling: props.id,
+          document,
+        })
+      )
+    }
+
+    store.dispatch(removeChild({ parent: parentPlugin.id, child: props.id }))
```

Comment:
Seems to work. I am fine with deploying it, when it is tested in staging very well. Can we only allow the Paste-Plugin for certain users? (e.g. Anne, Claudia, ...)

Reply:
@elbotho Can you test what happens, when a nested plugin is inserted (e.g. there is a box in the inserted data)

Reply:
> @elbotho Can you test what happens, when a nested plugin is inserted (e.g. there is a box in the inserted data)

Yeah, good idea.

Reply:
> Can we only allow the Paste-Plugin for certain users? (e.g. Anne, Claudia, ...)

I would put the plugin into an experimental feature anyway (this way only power-authors will find it) and we could also only show this experiment for selected roles or users.

Reply:
It's now behind a feature flag and also forbidden in production for now.